API workflow: generate API module, remove older README,

bobleesj opened the pull request:
<img width="1093" alt="Screenshot 2024-08-09 at 2 07 20 PM" src="https://github.com/user-attachments/assets/a95e8b16-b76c-464c-8d62-7e93a465e95c">

```bash
python ../../release-scripts/auto_api.py diffpy.snmf ./src/diffpy/snmf ./doc/source/api
python -m build
```

@sbillinge seems to be working. Please review.

sbillinge commented on setup.py:
```diff
@@ -16,7 +16,7 @@
 # with open(os.path.join(MYDIR, 'requirements/run.txt')) as fp:
```
I believe setup.py should be deleted.

sbillinge reviewed (commented):
please see inline comment.  Also, rever.xsh needs to be deleted.

bobleesj commented:
`setup.py` `rever.xsh ` removed. @sbillinge

bobleesj commented:
Hm, the `CI` seems to be failing. I will double check.

bobleesj commented:
https://github.com/bobleesj/diffpy.snmf/actions/runs/10325248674

Removed `setup.py` and modified `pyprojet.toml` (cross-checked diffpy.structure). 

CI passes. Please review. @sbillinge
